Added text align feature to cover image.

jorgefilipecosta opened the pull request:
## Description
This PR adds the option to change text-align in the cover image.

## How Has This Been Tested?
See cover images created before this change still work as expected.
Change the text algin of cover image blocks to left and right and see if things are ok.



## Screenshots (jpeg or gifs if applicable):
<img width="971" alt="screen shot 2017-12-18 at 14 59 20" src="https://user-images.githubusercontent.com/11271197/34112316-1da62ac2-e404-11e7-89f8-8633773cc6de.png">

jorgefilipecosta committed:
Added text align feature to cover image.

jasmussen commented:
Love it, nice! Works well!

We'll probably want to "right float" the block toolbar too, when the text is right aligned, otherwise on mobile we run into this:

![screen shot 2017-12-21 at 16 51 12](https://user-images.githubusercontent.com/1204802/34263296-3c5798da-e66f-11e7-9716-37d7b74b1d2a.png)

Nice work!

mcsf commented on blocks/test/fixtures/core__cover-image.json:
```diff
@@ -9,7 +9,8 @@
             "hasParallax": false,
-            "dimRatio": 40
+			"dimRatio": 40,
+			"contentAlign": "center"
```
Wrong indentation here.

mcsf reviewed (commented):
Looks good, but we should handle front-end styles:

<img width="751" alt="screen shot 2017-12-21 at 17 15 33" src="https://user-images.githubusercontent.com/150562/34267197-ea81bebe-e673-11e7-90f5-1407fdf2c40f.png">

Both the Heading and the Cover Image blocks there are set to align to the right. The issue is that `<h2>` can have `margin-right: auto` (as part of `margin: 1.5em auto` in the case of `gutenberg-theme`). In that Cover Image, the `<h2>` takes up the entire width of the `<section>` that contains it, hence its central positioning.

If we could fix this without resorting to a hard reset of `margin-right`, that'd be great, as we'd allow authors to customize that property even in Cover Image. Ideally, you'd get flex to not let `<h2>` grow. If not, we can fix it the hard way `¯\_(ツ)_/¯`.

mtias commented on blocks/library/cover-image/editor.scss:
```diff
@@ -22,4 +22,16 @@
 		color: inherit;
 	}
+
+	&.contentalignleft {
```
Let's use classes like `has-left-content` here as it's easier to control and be consistent (modifier classes like `is-` and `has-`) when we apply to `.block.has-left-content`.

mcsf replied:
I kind of assumed Jorge was keeping with the WordPress convention of `alignleft` _et al._. Should we push for change in that area too?

mtias replied:
I have left the alignleft / right / center untouched for now, but any new class we should take care of it. (Also applies to columns as something we need to address.)

mtias commented on blocks/library/cover-image/index.js:
```diff
@@ -139,6 +145,13 @@ registerBlockType( 'core/cover-image', {
 					max={ 100 }
 					step={ 10 }
 				/>
+				<p>{ __( 'Text Alignment' ) }</p>
```
Should this be an h2 as well?

mtias commented:
Looking good to me.

jorgefilipecosta commented:
Hi @jasmussen, @mcsf, @mtias thank you for your reviews, the points raised were addressed :)

jasmussen commented:
This is working nicely. 👍 👍  from me

Not necessary for this PR, but something to think about. Should we just show the alignment buttons in the inline toolbar, so you don't have to go into the Settings for the block?

![screen shot 2018-01-09 at 12 26 05](https://user-images.githubusercontent.com/1204802/34718785-568865b8-f538-11e7-91a6-3ce29fbe8bed.png)

... or does that impact the inline toolbar elsewhere?

mcsf commented on blocks/library/cover-image/editor.scss:
```diff
@@ -22,4 +22,16 @@
+		.block-editable__inline-toolbar{
+			justify-content: flex-end;
+		}
+	}
```
Minor, but we don't really need the nesting here `(&.has-left-content .block-editable__… {`).

mcsf commented on blocks/library/cover-image/index.js:
```diff
@@ -216,6 +231,7 @@ registerBlockType( 'core/cover-image', {
 				'has-parallax': hasParallax,
 			},
 			align ? `align${ align }` : null,
+			contentAlign !== 'center' ? `has-${ contentAlign }-content` : null,
```
The equivalent expression in `edit()` is a bit nicer IMO, with `&&`.

mcsf reviewed (approved):
Added minor comments, but not blockers. Looks good. :)

jorgefilipecosta commented:
I addressed @mcsf and I added alignment controls in the toolbar as suggested by @jasmussen. 
@karmatosed would it be possible to have a final look at the design before the merge :)?

karmatosed reviewed (approved):
I think this works well, thanks.
